server/src/test/java/io/spine/examples/shareaware/server/e2e/given/E2EUser.java

```diff
@@ -0,0 +1,314 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.extensions.proto.ProtoTruth.assertThat;
+import static com.google.common.util.concurrent.Uninterruptibles.sleepUninterruptibly;
+import static io.spine.client.Client.usingChannel;
+import static io.spine.examples.shareaware.MoneyCalculator.*;
+import static io.spine.examples.shareaware.given.GivenMoney.usd;
+import static io.spine.examples.shareaware.server.e2e.given.E2EUserTestEnv.purchaseSharesFor;
+import static io.spine.examples.shareaware.server.e2e.given.E2EUserTestEnv.replenishWallet;
+import static io.spine.examples.shareaware.server.e2e.given.E2EUserTestEnv.withdrawMoneyFrom;
+import static io.spine.examples.shareaware.server.e2e.given.SharePurchaseTestEnv.walletBalanceWith;
+import static io.spine.examples.shareaware.server.e2e.given.SharePurchaseTestEnv.zeroWalletBalance;
+import static io.spine.examples.shareaware.server.given.GivenWallet.createWallet;
+import static io.spine.util.Exceptions.illegalStateWithCauseOf;
+import static java.time.Duration.ofMillis;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Represents a user for end-to-end ShareAware application tests which interacts
+ * with the server-side with the help of {@link Client}.
+ */
+public final class E2EUser {
+
+    private final Client client;
+    private final UserId userId;
+    private final WalletId walletId;
+    private final WalletBalanceSubscription wallet;
+
+    public E2EUser(ManagedChannel channel) {
+        this.client = usingChannel(channel).build();
+        this.userId = GivenUserId.generated();
+        this.walletId = GivenWallet.walletId(userId);
+        wallet = new WalletBalanceSubscription(client, userId);
+        createWalletForUser();
+    }
+
+    /**
+     * Allows user to send the provided command to the server.
+     */
+    public void command(CommandMessage command) {
+        client.onBehalfOf(userId)
+              .command(command)
+              .postAndForget();
+    }
+
+    /**
+     * Returns the ID of the user.
+     */
+    public UserId id() {
+        return userId;
+    }
+
+    /**
+     * Returns the ID of the user's wallet.
+     */
+    public WalletId walletId() {
+        return walletId;
+    }
+
+    /**
+     * Describes the user's action to wait for shares to update on the market.
+     */
+    public List<Share> waitsForSharesToUpdate() {
+        sleepUninterruptibly(ofMillis(1500));
+        List<Share> shares = looksAtShares();
+        return shares;
+    }
+
+    /**
+     * Describes the user's action to replenish his wallet.
+     */
+    public WalletBalance replenishesWalletFor(Money amount) {
+        ReplenishWallet replenishWallet = replenishWallet(walletId, amount);
+
+        SubscriptionOutcome<WalletBalance> actualBalance =
```

Comment:
This variable is not used.

Please pay special attention to the warnings provided by IDEA inspections. All of them must be addressed prior to submitting the PR.